pmd-core/src/main/java/net/sourceforge/pmd/lang/document/NioTextFile.java

```diff
@@ -76,13 +76,18 @@ public void writeContents(TextFileContent content) throws IOException {
             throw new ReadOnlyFileException(this);
         }
         try (BufferedWriter bw = Files.newBufferedWriter(path, charset)) {
-            if (TextFileContent.NORMALIZED_LINE_TERM.equals(content.getLineTerminator())) {
-                content.getNormalizedText().writeFully(bw);
+            TextFileContent content = getContent();
+            if (content.hasNormalizedLineTerminator()) {
+                content.writeNormalizedTextFully(bw);
             } else {
-                for (Chars line : content.getNormalizedText().lines()) {
-                    line.writeFully(bw);
-                    bw.write(content.getLineTerminator());
-                }
+                NormalizedText normalizedText = content.getNormalizedText();
+                LineTerminator lineTerminator = content.getLineTerminator();
+                for (Chars line : normalizedText.lines()) {
```

Comment:
This will just plainly not compile.  You didn't commit a class `NormalizedText` or `LineTerminator`, not that they are needed since the current types `Chars` and `String` are appropriate. Similarly, you didn't commit `hasNormalizedLineTerminator`, which would maybe have been the only positive change here.